include/swift/AST/Decl.h

```diff
@@ -4995,7 +4995,13 @@ class AbstractFunctionDecl : public GenericContext, public ValueDecl {
   union {
     /// This enum member is active if getBodyKind() is BodyKind::Parsed or
     /// BodyKind::TypeChecked.
-    BraceStmt *Body;
+    struct {
+      /// The AST node of the body.
+      BraceStmt *Node;
+
+      /// The string representation of the body, if anything.
+      StringRef StringRepresentation;
```

Comment:
Hm. Aren't these going to be mutually exclusive, like they are for default argument values?

Reply:
I guess I should add `BodyKind::Deserialized` to cover that, but yes it should be mutually exclusive. The check for this would have to be outside of Parsed/TypeChecked